### PR TITLE
Add optional references

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  |  -

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,24 +2,25 @@
 
 ## 2.0.1 under development
 
-- no changes in this release.
+- Enh #43: Add `Reference::optional()` method that returns `null` on resolve dependencies when container don't have
+  entry. (@vjik)
```
Comment (samdark) [inline CHANGELOG.md:6]:
```suggestion
- Enh #43: Add `Reference::optional()` method that returns `null` when there is no dependency defined in container (@vjik)
```

File: README.md [hunk 2]
```diff
@@ -144,6 +144,20 @@ dependencies:
 ]
 ```
 
+Optional references returns `null` when container don't have entry:
```
Comment (samdark) [inline README.md:147]:
```suggestion
Optional reference returns `null` when there's no corresponding definition in container:
```

File: README.md [hunk 3]
```diff
@@ -144,6 +144,20 @@ dependencies:
...
+
+```php
+[
+    MyService::class => [
+        '__construct()' => [
+            // If container don't have implementation of `EventDispatcherInterface` then reference returns `null`
+            // when resolve dependencies
```
Comment (samdark) [inline README.md:154]:
```suggestion
            // If container doesn't have definition for `EventDispatcherInterface` reference returns `null`
            // when resolving dependencies 
```

File: src/Reference.php [hunk 4]
```diff
@@ -29,26 +29,43 @@
...
     {
-        if (!is_string($id)) {
-            throw new InvalidConfigException('Reference ID must be string.');
-        }
+        return new self($id, false);
+    }
 
-        return new self($id);
+    /**
+     * @param mixed $id ID of the service or object to point to.
```
Comment (samdark) [inline src/Reference.php:58]: Need phpdoc describing method purpose.

**Review by samdark — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (vjik) [reply CHANGELOG.md]: done

File: README.md (hunk 3 above)
Comment (vjik) [reply README.md]: done

File: src/Reference.php (hunk 4 above)
Comment (vjik) [reply src/Reference.php:60]: done

**Pull request merged**